cpp/include/raft/linalg/strided_reduction.cuh

```diff
@@ -71,6 +74,84 @@ void stridedReduction(OutType* dots,
   detail::stridedReduction(dots, data, D, N, init, stream, inplace, main_op, reduce_op, final_op);
 }
 
+/**
+ * @defgroup strided_reduction Strided Memory Access Reductions
+ * For reducing along rows for row-major and along columns for column-major
+ * @{
+ */
+
+/**
+ * @brief Compute reduction of the input matrix along the strided dimension
```

Comment:
Can you expand upon this description a little bit? These primitives have been infamous for not being the most descriptive and if we can provide more information, I think they will reach users who may have avoided using them due to lack of clarity in the docs.

Reply:
Also we've been trying to mark `[in]` and `[out]` in the docs and it's super useful to keep that consistent especially for our new public API functions.